### ISSUE TITLE
Improving the Xtext web page. 

### ISSUE THREAD

**Issue by miklossy** [eclipse/xtext#1731]
The following sections are currently missing and should be added to the [Eclipse Support](https://www.eclipse.org/Xtext/documentation/310_eclipse_support.html) web page:

- [x] Folding: [PR](https://github.com/eclipse/xtext/pull/1740) has been created and merged.
- [x] Hovering: [PR](https://github.com/eclipse/xtext/pull/1743) has been created and merged.
- [x] Autoediting: [PR](https://github.com/eclipse/xtext/pull/1744) has been created and merged.
- [x] Mark Occurrences: [PR](https://github.com/eclipse/xtext/pull/1757) has been created.
- [x] Find References: [PR](https://github.com/eclipse/xtext/pull/1757) has been created.

**Comment by miklossy** [comment]
As reported by @cdietrich , there are currently some invalid hyperlinks on the webpage. These should be fixed as well:
```
Status	URL	Source link text
-1 Timeout	http://live.eclipse.org/	Eclipse Live
404 Not Found	https://www.packtpub.com/sites/default/files/4965OS_5541_Implementing Domain Specific Languages with Xtext and Xtend â Second Edition.jpg	img/src
404 Not Found	https://github.com/eclipse/xtext-core/blob/master/org.eclipse.xtext.xtext.generator/src/org/eclipse/xtext/xtext/generator/model/project/SubProjectConfig.xtend	SubProjectConfig
404 Not Found	https://github.com/eclipse/xtext-core/blob/master/org.eclipse.xtext.xtext.generator/src/org/eclipse/xtext/xtext/generator/DefaultGeneratorModule.xtend	DefaultGeneratorModule
404 Not Found	https://github.com/eclipse/xtext-core/blob/master/org.eclipse.xtext.xtext.generator/src/org/eclipse/xtext/xtext/generator/XtextGeneratorNaming.xtend	XtextGeneratorNaming
404 Not Found	https://github.com/eclipse/xtext-core/blob/master/org.eclipse.xtext.xtext.generator/src/org/eclipse/xtext/xtext/generator/IXtextGeneratorFragment.xtend	generator fragments
-1 Timeout	http://logging.apache.org/log4j/	Apache Log4j
404 Not Found	https://github.com/eclipse/xtext-core/blob/master/org.eclipse.xtext.ide/src/org/eclipse/xtext/ide/editor/contentassist/IdeCrossrefProposalProvider.xtend	IdeCrossrefProposalProvider
404 Not Found	https://github.com/eclipse/xtext-core/blob/master/org.eclipse.xtext.ide/src/org/eclipse/xtext/ide/editor/contentassist/IdeContentProposalProvider.xtend	IdeContentProposalProvider
404 Not Found	https://github.com/eclipse/xtext-core/blob/master/org.eclipse.xtext.ide/src/org/eclipse/xtext/ide/editor/contentassist/ContentAssistEntry.xtend	ContentAssistEntry
404 Not Found	https://github.com/eclipse/xtext-web/blob/master/org.eclipse.xtext.web/src/main/java/org/eclipse/xtext/web/server/occurrences/OccurrencesService.xtend	OccurrencesService
404 Not Found	https://github.com/eclipse/xtext-core/blob/master/org.eclipse.xtext.ide/src/org/eclipse/xtext/ide/labels/INameLabelProvider.xtend	INameLabelProvider
404 Not Found	https://github.com/eclipse/xtext-core/blob/master/org.eclipse.xtext.ide/src/org/eclipse/xtext/ide/labels/IImageDescriptionProvider.xtend	IImageDescriptionProvider
404 Not Found	https://github.com/eclipse/xtext-web/blob/master/org.eclipse.xtext.web/src/main/java/org/eclipse/xtext/web/server/persistence/FileResourceHandler.xtend	FileResourceHandler
404 Not Found	https://github.com/eclipse/xtext-eclipse/blob/master/org.eclipse.xtext.ui/src/org/eclipse/xtext/ui/wizard/template/ProjectTemplate.xtend	ProjectTemplate
404 Not Found	https://github.com/eclipse/xtext-eclipse/blob/master/org.eclipse.xtext.ui/src/org/eclipse/xtext/ui/wizard/template/FileTemplate.xtend	FileTemplate
404 Not Found	https://github.com/eclipse/xtext-eclipse/blob/master/org.eclipse.xtext.ui.testing/src/org/eclipse/xtext/ui/testing/AbstractContentAssistTest.xtend	AbstractContentAssistTest
404 Not Found	https://github.com/eclipse/xtext-eclipse/blob/master/org.eclipse.xtext.ui.testing/src/org/eclipse/xtext/ui/testing/AbstractFoldingTest.xtend	AbstractFoldingTest
404 Not Found	https://github.com/eclipse/xtext-eclipse/blob/master/org.eclipse.xtext.ui.testing/src/org/eclipse/xtext/ui/testing/AbstractHighlightingTest.xtend	AbstractHighlightingTest
404 Not Found	https://github.com/eclipse/xtext-eclipse/blob/master/org.eclipse.xtext.ui.testing/src/org/eclipse/xtext/ui/testing/AbstractHoverTest.xtend	AbstractHoverTest
404 Not Found	https://github.com/eclipse/xtext-eclipse/blob/master/org.eclipse.xtext.ui.testing/src/org/eclipse/xtext/ui/testing/AbstractHyperlinkingTest.xtend	AbstractHyperlinkingTest
404 Not Found	https://github.com/eclipse/xtext-eclipse/blob/master/org.eclipse.xtext.ui.testing/src/org/eclipse/xtext/ui/testing/AbstractQuickfixTest.xtend	AbstractQuickfixTest
404 Not Found	http://download.eclipse.org/eclipse/updates/4.5-M-builds/	4.5.2
999 Non-standard	https://www.linkedin.com/company/eclipse-foundation	<No Text>
404 Not Found	http://www.ancitconsulting.com/tiki-index.php?page=Tutorials	Tutorials by ANCiT Consulting
404 Not Found	https://projects.eclipse.org/projects/rt.concierge/downloads	Get it
-1 Timeout	https://maven.apache.org/guides/introduction/introduction-to-dependency-mechanism.html	Maven BOM (bill of materials)
-1 Not found: The server name or address could not be resolved 	http://new.axdt.org/	Axdt
-1 Not found: The host name in the certificate is invalid or does not match 	https://balzac-lang.xyz/	Online editor
-1 Not found: The host name in the certificate is invalid or does not match 	https://docs.balzac-lang.xyz/	documentation
-1 Not found: The server name or address could not be resolved 	http://businessAppTester.com/	BusinessAppTester
-1 Not found: An error occurred in the secure channel support 	https://graphviz.gitlab.io/_pages/doc/info/lang.html	Graphviz DOT language
-1 Not found: The server name or address could not be resolved 	http://metrikforge.informatik.hu-berlin.de/projects/nanoworkbench/	NanoWorkbench
-1 Not found: The server name or address could not be resolved 	http://www.gipsys.ca/xdgl.htm	xDGL Data Generation Language
429 Too Many Requests	https://github.com/eclipse/xtext-lib/blob/master/org.eclipse.xtend.lib/src/org/eclipse/xtend/lib/annotations/Delegate.xtend	@Delegate
429 Too Many Requests	https://github.com/eclipse/xtext/edit/website-published/xtend-website/documentation/index.md	Edit on Github
429 Too Many Requests	https://github.com/eclipse/xtext-xtend/issues	<No Text>
429 Too Many Requests	https://github.com/eclipse/xtext/edit/website-published/xtend-website/documentation/101_gettingstarted.md	Edit on Github
429 Too Many Requests	https://github.com/eclipse/xtext-lib/blob/master/org.eclipse.xtext.xbase.lib/src/org/eclipse/xtext/xbase/lib/InputOutput.java	InputOutput
429 Too Many Requests	https://github.com/eclipse/xtext/edit/website-published/xtend-website/documentation/201_types.md	Edit on Github
429 Too Many Requests	https://github.com/eclipse/xtext/edit/website-published/xtend-website/documentation/102_moviesexample.md	Edit on Github
404 Not Found	http://docs.guava-libraries.googlecode.com/git-history/release/javadoc/com/google/common/io/CharStreams.html	CharStreams
429 Too Many Requests	https://github.com/eclipse/xtext-lib/blob/master/org.eclipse.xtext.xbase.lib/src/org/eclipse/xtext/xbase/lib/IntegerRange.java	IntegerRange
404 Not Found	https://eclipsecon.us6.list-manage.com/subscribe/post	form/action(post)
404 Not Found	https://www.eclipse.org/js/solstice.hugo.js	script/src
404 Not Found	https://www.eclipse.org/Xtext/version-2-9-0	new generator infrastructure
404 Not Found	https://www.eclipse.org/Xtext/kermeta.org/	Kermeta 3
404 Not Found	https://www.eclipse.org/xtend/media/Xtend_intro_720p.mp4	source/src
```

**Comment by miklossy** [comment]
Fixed in 2.22
